app/src/main/java/com/github/sdp_begreen/begreen/activities/SignInActivity.kt

```diff
@@ -56,7 +56,15 @@ class SignInActivity : AppCompatActivity() {
         signInGoogleLayout =findViewById(R.id.signInGoogleLayout)
         firebaseAuth = FirebaseAuth.getInstance()
 
-        setUpSignInLayout()
+        // Check if firebaseAuth.currentUser is not null which means that User is already logged in
+        // This allow offline mode authentication
+        if(firebaseAuth.currentUser!=null) {
+            startActivity(Intent(this@SignInActivity, MainActivity::class.java))
+            finish()
+        }
+        else {
+            setUpSignInLayout()
+        }
```

Comment:
I see we are still using `FirebaseAuth` directly in this class, instead of using injections. Even if this should not change anything to use it here, I think it is good practice to be consistend with the rest of the code. To use the injection of firebaseAuth, we should remove the var `firebaseAuth` at the beginning of the file and replace it with `private val auth by inject<Auth>()`, and use this variable in the file instead. It will also probably make tests easier to write. Note that we would still need to add a new method to `Auth` and `FirebaseAuth` to log the user in, tell me if you have any doubts!

Reply:
Thank you for your comment, Luca! Do you happen to have an example of this injection in our code or elsewhere that I could use as a reference for my own code?

Reply:
Of course,
In the `MainActivity` by example, we wrote
`private val auth by inject<Auth>()` 
on line 40.
Then we used this variable to make calls to `AuthDB`.
Is it clearer?